diff --git a/embed/build.sh b/embed/build.sh
index 014bd22..f9ccc7d 100755
--- a/embed/build.sh
+++ b/embed/build.sh
@@ -21,6 +21,7 @@ WASI_SDK="$ROOT/tools/wasi-sdk-23.0/bin"
 	-Wl,--initial-memory=327680 \
 	-D_HAVE_SQLITE_CONFIG_H \
 	-DSQLITE_CUSTOM_INCLUDE=sqlite_opt.h \
+  -DSQLITE_VEC_OMIT_FS=1 \
 	$(awk '{print "-Wl,--export="$0}' exports.txt)
 
 trap 'rm -f sqlite3.tmp' EXIT
diff --git a/sqlite3/main.c b/sqlite3/main.c
index 4fb926b..4cade42 100644
--- a/sqlite3/main.c
+++ b/sqlite3/main.c
@@ -16,7 +16,7 @@
 #include "time.c"
 #include "vfs.c"
 #include "vtab.c"
-
+#include "../../sqlite-vec.c"
 sqlite3_destructor_type malloc_destructor = &free;
 
 __attribute__((constructor)) void init() {
@@ -28,4 +28,5 @@ __attribute__((constructor)) void init() {
   sqlite3_auto_extension((void (*)(void))sqlite3_series_init);
   sqlite3_auto_extension((void (*)(void))sqlite3_uint_init);
   sqlite3_auto_extension((void (*)(void))sqlite3_time_init);
+  sqlite3_auto_extension((void (*)(void))sqlite3_vec_init);
 }
\ No newline at end of file
